### DISCUSSION TITLE
--no-auto-back-and-forth behaviour

### DISCUSSION THREAD

**Discussion by qrockz** [i3/i3#5996] [Q&A] [answered]
For instance: 

- previous focused ws: 7
- current focused ws: 10

I want to switch to `ws 2` and back to `ws 10` without breaking the `workspace back_and_forth` toggle between `ws 7` and `ws 10`.

`i3-msg "workspace --no-auto-back-and-forth number 2; workspace --no-auto-back-and-forth number 10"`

 
I expect `workspace back_and_forth`  still be able to toggle between `ws 7` and `ws 10` because of the  `--no-auto-back-and-forth` flag I've used for switching to `ws 2`.
But now `workspace back_and_forth `toggles between` ws 2` and `ws 10 `instead of `ws 7` and `ws 10` :confused: .

**Originally I wanted to change  the workspace on another output/monitor without losing the focus on the current used monitor.**
I.e.I'm on output a and want the active workspace on output b changed without moving the focus to  the new workspace on output b.  Is that possible? Any ideas? 

This script changes workspaces on another outpud HDMI-1 and also preserves the current back_and_forth functionality. 
The downside is that this is not efficient due to flickering when using  `[...] workspace "$ws_b"; workspace "$ws_a"`.
```
ws_a=$(i3-msg -t get_workspaces | jq -r '.[] |select(.focused==true)| .name')
i3-msg "workspace back_and_forth"
ws_b=$(i3-msg -t get_workspaces | jq -r '.[] |select(.focused==true)| .name')
i3-msg "focus output HDMI-1; workspace next_on_output; workspace "$ws_b"; workspace "$ws_a""
```

i3 --version
i3 version 4.23

**Comment by orestisfl** [comment]
`workspace --no-auto-back-and-forth number 2` means that if workspace number 2 is currently focused, you will not switch to the previously focused workspace even if `workspace_auto_back_and_forth yes` is set in your config.

**Reply by qrockz** [reply]
@orestisfl 

Is it possible to change workspaces on  _displays b_ without losing the current focused _display a_ ?

_Moving_ windows to other workspaces without following the focus works (ws 2 has focus. move window from ws 2 to ws 3. ws 2 still has focus). 
It would be nice to have this behaviour also for changing workspaces or even containers on other displays without losing the focus on the current display :pray:

**Reply by orestisfl** [reply, marked as answer]
So, for example workspaces `1` and `2` are on display a and `1` is focused and workspaces `3` and `4` are on display b and `3` is visible. You would like to make `4` visible without losing focus on `1`? Then yes, that's not possible.
